storm-core/test/jvm/backtype/storm/utils/RateTrackerTest.java

```diff
@@ -27,36 +27,63 @@
 public class RateTrackerTest extends TestCase {
 
     @Test
+    public void testExactRate() {
+        final long interval = 1000l;
+        long time = 0l;
+        RateTracker rt = new RateTracker(10000, 10, time);
+        double [] expected = new double[] {10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0};
+        for (int i = 0; i < expected.length; i++) {
+            double exp = expected[i];
+            rt.notify(10);
+            time += interval;
+            double actual = rt.reportRate(time);
+            rt.forceRotate(1, interval);
+            assertEquals("Expected rate on iteration "+i+" is wrong.", exp, actual, 0.00001);
+        }
+        expected = new double[] {11.0, 12.0, 13.0, 14.0, 15.0, 16.0, 17.0, 18.0, 19.0, 20.0};
```

Comment:
It would be better to comment that it is continuous test after previous loop, so for each loop earliest bucket (which stores 10) is eclipsed and replaced to 20.
I was misunderstood L43-51 to a new test, so I spent some times to find out why it is not 20.0.